tests/test/existential_types.rs

```diff
@@ -2,6 +2,33 @@
 
 use super::*;
 
+#[test]
+fn opaque_bounds() {
+    test! {
+        program {
+            trait Trait { }
+            struct Ty { }
+            impl Trait for Ty { }
+
+            opaque type T: Trait = Ty;
+        }
+
+        goal {
+            if (Reveal) {
+                T: Trait
+            }
+        } yields {
+            "Unique; substitution []"
+        }
+
+        goal {
+            T: Trait
+        } yields {
+            "No possible solution"
```

Comment:
ok this doesn't seem right =) shouldn't this be solveable, since the bounds say `T: Trait`?

This could well be related to the 'placeholder' thing I was talking about above

Reply:
Right, I've updated the test. But what about a situation where we have a hidden type that doesn't actually implement the bound trait? Do we just assume it?

Reply:
nvm, that's what the WF rules are for of course